File: Part 2 - Regression/Section 9 - Random Forest Regression/random_forest_regression.R
# Random Forest Regression
# install.packages("randomForest")
# setwd("~/work/mlatoz/Part 2 - Regression/Section 9 - Random Forest Regression")
library(randomForest)
library(ggplot2)
# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


# ====> Random forest regression
set.seed(123) # ====> for random factor to be fixed 
regressor = randomForest(dataset[1], y = dataset$Salary, ntree = 500)

predict(regressor, newdata = data.frame(Level = 6.5))

# ====> Visualizing the result of Random forest regression
level_grid = seq(min(dataset$Level), max(dataset$Level), by = 0.01)
exp_plot = ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
  geom_line(aes(x = level_grid, y = predict(regressor, newdata = data.frame(Level = level_grid))), color = 'blue') +
  ggtitle('Level vs Salary (Random forest regression)') + 
  xlab('Level') +
  ylab('Salary')

ggsave(filename = 'r_random_forest.png', plot = exp_plot)

File: Part 2 - Regression/Section 9 - Random Forest Regression/random_forest_regression.py
# Random Forest Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.ensemble import RandomForestRegressor

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
level = dataset.iloc[:, 1:2].values
salary = dataset.iloc[:, 2].values


# ====> Random forest regressor 
regressor = RandomForestRegressor(n_estimators = 300, random_state = 0)
regressor.fit(level, salary)

regressor.predict(6.5)

# ====> Plot Random forest Regression visualization
level_grid = np.arange(min(level), max(level), step = 0.01)
level_grid = level_grid.reshape(len(level_grid), 1)
plt.scatter(level, salary, color = 'red')
plt.plot(level_grid, regressor.predict(level_grid), color = 'blue')
plt.title('Level vs Salary (Random forest regressor)')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.savefig('py_random_forest.png')
plt.close()